### DISCUSSION TITLE
Add flag to directus-extension to disable the terser plugin.

### DISCUSSION THREAD

**Discussion by miccoh1994** [directus/directus#8793] [Feature Requests]
Sometimes it is difficult to debug issues when developing extensions because of the terser plugin. It would be great if there was an option for the directus-extension cli like `-d` or `--dev` to disable terser when developing. I always have to change it manually in node_modules...

**Comment by nickrum** [comment]
We are planning to add a dedicated `directus-extension dev` command with full HMR and support for Vue Devtools. That should improve the debugging experience a lot. The `directus-extension build` command is actually only meant for production builds, which should always be minified.

I was thinking about adding an option to the extension config to be able to disable certain internal plugins. That would help with your issue until the `dev` command is implemented.

**Comment by Voldemorten** [comment]
If anyone needs a quick workaround, you can use the build feature directly from node_modules, which accepts a parameter -m that enables source maps. This gives you the unminified file. 
I suggest adding this to your package.json

```
"scripts": {
    "build": "directus-extension build",
    "build-dev": "node ./node_modules/.bin/directus-extension build -m"
}
```

**Reply by nickrum** [reply]
Using `directus-extension build --sourceMaps` should actually work perfectly fine. No need to import the CLI binary from `node_modules` 😉

Something to look out for: The next release of Directus will split up the `-m, --sourceMaps` flag into dedicated `--no-minify` and `--sourcemap` flags to align with other tools like rollup and vite.

**Comment by nickrum** [comment]
A `--no-minify` flag has since been added to the Extensions CLI.
